source/MRMesh/MRConeObject.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include "MRMeshFwd.h"
+#include "MRObjectMeshHolder.h"
+
+namespace MR
+{
+
+/// Object to show Cylinder feature, position and radius are controlled by xf
+/// \ingroup FeaturesGroup
+class MRMESH_CLASS ConeObject : public ObjectMeshHolder
+{
+public:
+    /// Creates simple Cylinder object with center in zero and radius - 1
+    MRMESH_API ConeObject();
+    /// Finds best Cylinder to approx given points
+    MRMESH_API ConeObject( const std::vector<Vector3f>& pointsToApprox );
+
+    ConeObject( ConeObject&& ) noexcept = default;
+    ConeObject& operator = ( ConeObject&& ) noexcept = default;
+
+    constexpr static const char* TypeName() noexcept
+    {
+        return "ConeObject";
+    }
+    virtual const char* typeName() const override
+    {
+        return TypeName();
+    }
+
+    /// \note this ctor is public only for std::make_shared used inside clone()
+    ConeObject( ProtectedStruct, const ConeObject& obj ) : ConeObject( obj )
+    {}
+
+    virtual std::string getClassName() const override
+    {
+        return "Cylinder";
+    }
+
+    MRMESH_API virtual std::shared_ptr<Object> clone() const override;
+    MRMESH_API virtual std::shared_ptr<Object> shallowClone() const override;
+
+    /// calculates radius from xf
+    MRMESH_API float getAngle() const;
+    /// calculates center from xf
+    MRMESH_API Vector3f getCenter() const;
+    /// updates xf to fit given radius
+    MRMESH_API void setAngle( float radius );
```

Comment:
1. Rename parameter to `angle`.

2. Add to the comment: is this the full angle or the half-angle?

Reply:
+